src/station_cmd.cpp

```diff
@@ -2562,9 +2562,9 @@ CommandCost CmdBuildDock(TileIndex tile, DoCommandFlag flags, uint32 p1, uint32
 	if (ret.Failed()) return ret;
 
 	tile_cur += TileOffsByDiagDir(direction);
-	if (!IsTileType(tile_cur, MP_WATER) || !IsTileFlat(tile_cur)) {
-		return_cmd_error(STR_ERROR_SITE_UNSUITABLE);
-	}
+
+	/* Ensure the location where ships dock, has water tracks. */
+	if (GetTileShipTrackStatus(tile_cur) == TRACK_BIT_NONE) return_cmd_error(STR_ERROR_SITE_UNSUITABLE);
```

Comment:
Should it not be `GetTileTrackStatus`, which is already available without needing to uninline things?

Reply:
I ended doing it this way. I couldn't grasp the concept of `inline`. Sorry, TrueBrain.